flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java

```diff
@@ -383,6 +394,25 @@
                     .withDescription(
                             "Configure the minimum increase in parallelism for a job to scale up.");
 
+    @Documentation.Section({
+        Documentation.Sections.EXPERT_SCHEDULING,
+        Documentation.Sections.ALL_JOB_MANAGER
+    })
+    public static final ConfigOption<Duration> RESOURCE_WAIT_TIMEOUT =
+            key("jobmanager.declarative-scheduler.resource-wait-timeout")
+                    .durationType()
+                    .defaultValue(Duration.ofSeconds(10))
+                    .withDescription(
+                            Description.builder()
+                                    .text(
+                                            "The maximum time the JobManager will wait for to acquire all required resources after a job submission or restart."
+                                                    + "Once elapsed it will try to run the job with a lower parallelism, or fail if the minimum amount of resources could not be acquired.")
```

Comment:
```suggestion
                                            "The maximum time the JobManager will wait to acquire all required resources after a job submission or restart. "
                                                    + "Once elapsed it will try to run the job with a lower parallelism, or fail if the minimum amount of resources could not be acquired.")
```